src/Hooks/CheckoutScriptHooks.php

```diff
@@ -199,7 +199,14 @@ private function shouldShowDeliveryOptions(): bool
             /**  @var WC_Product $product */
             $product = $cartItem['data'];
 
-            if (! $product->is_virtual() && ! $product->is_on_backorder($cartItem['quantity'])) {
+            $deliveryOptionsEnabledWhenNotInStock = Settings::get(
+                CheckoutSettings::ENABLE_DELIVERY_OPTIONS_WHEN_NOT_IN_STOCK,
+                CheckoutSettings::ID
+            );
+
+            if (! $product->is_virtual()
```

Comment:
deze if statement is slecht leesbaar, denk dat het helpt als je die backorder statement ook in een losse variabele zet